R/string_operation.R

```diff
@@ -19,21 +40,51 @@ is_digit <- function(word){
 #' @param word Character to be checked if it's digits.
 #' @return Logical vector if the word is digits or not.
 #' @export
-is_alphabet <- function(word, lexicon="snowball"){
+is_alphabet <- function(word){
   loadNamespace("stringr")
   stringr::str_detect(word, "^[[:alpha:]]+$")
 }
 
 #' Get vector of stopwords
-#' @param lexicon Type of stopwords. One of "snowball", "onix" and "SMART".
-#' @return vector of stop word.
+#' @param lexicon Type of stopwords.
+#' One of
+#' "danish",
+#' "dutch",
+#' "english",
+#' "finnish",
+#' "french",
+#' "german",
+#' "hungarian",
+#' "italian",
+#' "japanese",
+#' "norwegian",
+#' "portuguese",
+#' "russian",
+#' "spanish",
+#' "swedish",
+#' "japanese",
+#' "smart",
+#' "snowball",
+#' "onix"
+#' @param include Values that should be included as stop words
+#' @param exclude Values that should be excluded from stop words
+#' @return vector of stop words.
 #' @export
-get_stopwords <- function(lexicon="snowball"){
-  loadNamespace("tidytext")
-  data("stop_words", package = "tidytext", envir = environment())
-  type_check <- stop_words$lexicon %in% lexicon
-  words <- stop_words$word[type_check]
-  unique(words)
+get_stopwords <- function(lexicon = "english", include = c(), exclude = c()){
+  lexcon <- tolower(lexicon)
+  stopwords <- if (lexicon %in% c(
+    "snowball",
+    "onix",
+    "smart",
+    "japanese")){
+    get(paste0("stopwords_", lexicon))
```

Comment:
Comment that tell that those data are created in create_internal_data.R would help.